Minor cleanups, style tweaks

tkelman opened the pull request:
Mostly removing empty lines right at the start of a block or right before end, consolidating zero-arg `let`s in many of the tests to save a few lines, and eval-ing a deprecation from https://github.com/JuliaLang/julia/pull/19901 into the right module (that commit is best viewed as https://github.com/JuliaLang/julia/pull/20031/commits/f397cad6b2441658bb90ff0a277de6ad6122f2b8?w=1, as is the last commit https://github.com/JuliaLang/julia/pull/20031/commits/d5ea34842b1cb47dab4c2e603fd781bb805f0d6f?w=1 - or overall, https://github.com/JuliaLang/julia/pull/20031/files?w=1).

tkelman committed:
Save some vertical space with some previously zero-argument lets

tkelman committed:
remove a bunch of unnecessary blank lines

especially right after a block opener and right before end

tkelman committed:
none of the other bullets here pretend to be sentences

tkelman committed:
don't shadow cmp function with a variable name in test/worlds.jl

tkelman committed:
space try expression over multiple lines

tkelman committed:
Fix subj/verb agreement in broadcast docstring

tkelman committed:
wrap long lines in test/linalg/lu.jl

tkelman committed:
add back test examples for isapprox

todo: make this a doctest

tkelman committed:
Add a test that the <- operator from 19765 would have broken

tkelman committed:
fix grammar in a comment

tkelman committed:
replace a let with a testset

tkelman committed:
Prefer 'environment variable' spelling

over 'environmental variable'

tkelman committed:
4 space indent control flow examples

tkelman committed:
use logical negation instead of bitwise in triangular.jl

and linalg/dense.jl

tkelman committed:
can use deepcopy_internal without Base. qualification inside Base

tkelman committed:
Spell out FIFO acronym in channel documentation

tkelman committed:
eval Base.Test deprecations into right module

tkelman committed:
do not indent preprocessor conditionals in codegen.cpp

tkelman committed:
wrap some long lines in base/multidimensional.jl

stevengj commented on test/dict.jl:
```diff
@@ -651,7 +640,6 @@ Dict(1 => rand(2,3), 'c' => "asdf") # just make sure this does not trigger a dep
-
 @test hash(Dict(Dict(1=>2) => 3, Dict(4=>5) => 6)) != hash(Dict(Dict(4=>5) => 3, Dict(1=>2) => 6))
 let a = Dict(Dict(3 => 4, 2 => 3) => 2, Dict(1 => 2, 5 => 6) => 1)
     b = Dict(Dict(1 => 2, 2 => 3, 5 => 6) => 1, Dict(3 => 4) => 2)
```
Looks like there is a comma missing on the previous line, also (so that `b` is a `let` variable like `a`, not that it matters much.)

Sacha0 commented on base/linalg/dense.jl:
```diff
@@ -509,7 +509,7 @@ function logm(A::StridedMatrix)
     end
 
-    if isreal(A) && ~np_real_eigs
+    if isreal(A) && !np_real_eigs
```
:)

Sacha0 commented on base/multidimensional.jl:
```diff
@@ -51,10 +51,14 @@ module IteratorsMD
 
     # arithmetic, min/max
     (-){N}(index::CartesianIndex{N}) = CartesianIndex{N}(map(-, index.I))
-    (+){N}(index1::CartesianIndex{N}, index2::CartesianIndex{N}) = CartesianIndex{N}(map(+, index1.I, index2.I))
```
Do these definitions substantially exceed the line length recommendation? These eyes find the existing (aligned) version easier to parse as a whole than the line-split version. Best!

tkelman replied:
113 columns

Sacha0 commented on doc/src/manual/control-flow.md:
```diff
@@ -843,7 +843,8 @@ may have more values to generate and so might not yet be ready to return. With t
 Julia provides a [`Channel`](@ref) mechanism for solving this problem.
-A [`Channel`](@ref) is a waitable FIFO queue which can have multiple tasks reading and writing to it.
+A [`Channel`](@ref) is a waitable first-in first-out queue which can have
+multiple tasks reading and writing to it.
```
"reading and" -> "reading from and"?

Sacha0 commented on test/arrayops.jl:
```diff
@@ -904,8 +901,7 @@ end
 end
 
 @testset "single multidimensional index" begin
-    let
```
Is this `let` block redundant? (None of `a`, `a2`, `b`, or `I` occur in the enclosing scope?)

Sacha0 commented on test/arrayops.jl:
```diff
@@ -1853,11 +1847,10 @@ for op in (:.+, :.*, :.÷, :.%, :.<<, :.>>, :.-, :./, :.\, :.//, :.^)
 @testset "concatenations of dense matrices/vectors yield dense matrices/vectors" begin
-    let
-        N = 4
+    let N = 4
```
Likewise here, this `let` seems redundant with the `@testset`?

Sacha0 commented on test/complex.jl:
```diff
@@ -208,8 +208,7 @@ end
 @testset "log" begin
-    #  log(conj(z)) = conj(log(z))
-
+    # log(conj(z)) = conj(log(z))
```
Perhaps incorporate the comment in the testset description and nix the comment altogether?

Sacha0 commented on test/complex.jl:
```diff
@@ -242,8 +241,7 @@ end
 @testset "exp" begin
-    #  exp(conj(z)) = conj(exp(z))
-
+    # exp(conj(z)) = conj(exp(z))
```
Likewise here, perhaps incorporate the comment in the testset description and nix the comment?

Sacha0 commented on test/complex.jl:
```diff
@@ -278,8 +276,7 @@ end
 @testset "expm1" begin
-    #  expm1(conj(z)) = conj(expm1(z))
-
+    # expm1(conj(z)) = conj(expm1(z))
```
Likewise here, perhaps incorporate the comment in the testset description and nix the comment?

Sacha0 commented on test/complex.jl:
```diff
@@ -351,9 +348,9 @@ end
-    #  conj(x)^conj(y) = conj(x^y)
+    # equivalent to exp(y*log(x))
+    #   except for 0^0?
+    # conj(x)^conj(y) = conj(x^y)
```
Perhaps also a candidate for expanding the testset description and removing the comment?

Sacha0 commented on test/complex.jl:
```diff
@@ -489,8 +486,8 @@ end
-    #  tanh(conj(z)) = conj(tanh(z))
-    #  tanh(-z) = -tanh(z)
+    # tanh(conj(z)) = conj(tanh(z))
+    # tanh(-z) = -tanh(z)
```
Likewise here, perhaps incorporate the comments in the testset description and nix the comments?

Sacha0 commented on test/linalg/lu.jl:
```diff
@@ -29,8 +29,12 @@ dreal = randn(n)/2
+    d = eltya == Int ? Tridiagonal(rand(1:7, n-1), rand(1:7, n), rand(1:7, n-1)) :
+        convert(Tridiagonal{eltya}, eltya <: Complex ?
+        Tridiagonal(complex.(dlreal, dlimg), complex.(dreal, dimg), complex.(dureal, duimg)) :
+        Tridiagonal(dlreal, dreal, dureal))
```
Perhaps indent the two `Tridiagonal(...` lines, making their nesting within the `convert` more obvious?

Sacha0 commented on test/linalg/lu.jl:
```diff
@@ -167,7 +173,9 @@ end
-Hinv = Rational{BigInt}[(-1)^(i+j)*(i+j-1)*binomial(nHilbert+i-1,nHilbert-j)*binomial(nHilbert+j-1,nHilbert-i)*binomial(i+j-2,i-1)^2 for i = big(1):nHilbert,j=big(1):nHilbert]
+Hinv = Rational{BigInt}[(-1)^(i+j)*(i+j-1)*binomial(nHilbert+i-1,nHilbert-j)*
+    binomial(nHilbert+j-1,nHilbert-i)*binomial(i+j-2,i-1)^2
+    for i = big(1):nHilbert,j=big(1):nHilbert]
```
Perhaps indent the `binomial(...` and `for(...` lines further to clarify their part in the comprehension?

tkelman replied:
to where? it would start getting overly wide to align to the `[`

Sacha0 replied:
The `[` was my thought, yes.

tkelman replied:
I'm not much of a fan of horizontal-alignment games like that, I'd rather just apply a simple set of rules that we could conceivably automate some day.

Sacha0 commented on test/parse.jl:
```diff
@@ -384,6 +384,9 @@ end
 @test parse("x>:y<:z").head === :comparison
 
+# reason PR #19765, <- operator, was reverted
+@test -2<-1 # DO NOT ADD SPACES
```
`# DO NOT ADD SPACES` :laughing:

tkelman committed:
Address review comments

remove lets where redundant with testsets

move some comments into testset descriptions

ararslan commented:
Is this good to go?

tkelman commented:
As far as I'm concerned yes, though when merged it might cause conflicts with PR's that are messier to rebase than this would be.

tkelman commented:
thanks for reviewing!
